log4j-core/src/test/java/org/apache/logging/log4j/core/config/LoggerContextChangeTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache license, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the license for the specific language governing permissions and
+ * limitations under the license.
+ */
+
+package org.apache.logging.log4j.core.config;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.Assert;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.util.Set;
+
+public class LoggerContextChangeTest {
```

Comment:
When I run this test without the changes you have introduced in `AsyncAppender` and `LoggerContext`, it still passes. I don't think this test stresses any aspect of either `LoggerContext` or `AsyncAppender` that has changed via this PR. Am I missing something?

Reply:
But I get java.lang.AssertionError if loggercontext didn't change....

Reply:
Checking out _just_ `LoggerContextChangeTest.java` from your branch and running it against `release-2.x` simply succeeds for me:

```bash
git remote add wuqian0808 https://github.com/wuqian0808/logging-log4j2.git
git fetch wuqian0808
git checkout wuqian0808/LOG4J2-3182 -- log4j-core/src/test/java/org/apache/logging/log4j/core/config/LoggerContextChangeTest.java
./mvnw  -pl :log4j-core test -Dtest=LoggerContextChangeTest
```

Reply:
You should also use the confiugred log4j2-3182.xml and log4j2-3182-error.xml. Otherwise you won't get kind of this issue

Reply:
@wuqian0808, test succeeds anyway without your changes:

```bash
git remote add wuqian0808 https://github.com/wuqian0808/logging-log4j2.git
git fetch wuqian0808
git checkout wuqian0808/LOG4J2-3182 -- \
log4j-core/src/test/resources/log4j2-3182.xml \
log4j-core/src/test/resources/log4j2-3182-error.xml \
log4j-core/src/test/java/org/apache/logging/log4j/core/config/LoggerContextChangeTest.java
./mvnw clean package -DskipTests
./mvnw -pl :log4j-core test -Dtest=LoggerContextChangeTest
```

@wuqian0808, mind telling me what am I missing?